test/mw_shim.js

```diff
@@ -0,0 +1,849 @@
+/**
+ * Helper functions to fake mw object data for CLI testing of the Morebits library.
+ *
+ * Includes just what we care about:
+ * log
+ * Map
+ * config (barebones, will need data set)
+ * util (but plenty of excessive functions)
+ *     - All of morebits just needs: escapeRegExp, getParamValue, getUrl, isIPv6Address, wikiScript
+ *         - getParamValue (mb.w.page), wikiScript (mb.w.api), getUrl (batch, others)
+ *         - getUrl needs wikiScript, wikiUrlEncode, escapeIdForLink
+ *     - twinkle.js would need: addPortletLink
+ *     - modules would need: addCSS, isIPAddress, rawurlencode, wikiUrlencode
+ *
+ * Not funtional/done:
+ * user.tokens (nonfunctional, only used in Morebits.wiki.page)
+ * Title (only used in Morebits.wiki.page)
+ */
+
+var mw = {};
+
+/**
+ * mw.log
+ * We just need the `warn` method, but might as well mw.log = console.log
+ * node repl doesn't have console.warn
```

Comment:
? node repl does have it.

Reply:
Dunno WTF that was, I must've been overwriting it when first testing...